src/common_union.rs

```diff
@@ -111,6 +114,25 @@ impl FromIterator<Option<JsonUnionField>> for JsonUnion {
     }
 }
 
+/// Tries to collect an array represented as a string to a `JsonUnion`
+impl FromIterator<Option<String>> for JsonUnion {
```

Comment:
I don't really understand this, if we're just collecting some optional strings, how can we know they all represent arrays?

Reply:
I don't think this is the right way to go about what (I think) you're trying to do.

If you need this logic, better to put it where you need it.